### ISSUE TITLE
ARA fails when using too many hosts on --limit

### ISSUE THREAD

**Issue by rntbrgtt** [ansible-community/ara#185]
ARA 1.5.1
Installed from pip
Virtual Env
Debian GNU/Linux 10 (buster)
Mysql Database

## What is the issue ?

Whenever we run ansible with a lot of hosts on "--limit" there will be an error message and the output will not go to the DB.

To reproduce it simply run any playbook and add a lot of hosts on it's limit.

For example: ansible-playbook plabook.yml --limit 'A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z'

## What should be happening ?

ARA should work and execution output should go to the database.

**Comment by dmsimard** [comment]
Hey @rntbrgtt and thanks for reporting the issue.

Could you provide the error message you are seeing ? There might also be a full traceback available if you run ansible-playbook with ``-vvv``.

Thanks !

**Comment by rntbrgtt** [comment]
Hi David,

Thanks for your reply.

This is an old run so I dont have -vvv to show. 

```
xxx:~/xxx$ ansible-playbook playbooks/xxx.yml --check --diff -l 'xxx-1,xxx-2,xxx-3,xxx-4,xxx-5,xxx-6,xxx-7,xxx-8,xxx-9,xxx-10,xxx-11,xxx-12,xxx-13,xxx-14,xxx-15,xxx-16,17,18,19,20,!21,!22,!23,!24,!25,!26,!27,!28' -b

PLAY [Catch-all playbook] *******************************************************************************************************************************************************************************************************************
Failed to patch on /api/v1/playbooks/2175: {'labels': ['check:True', 'tags:all', 'subset:xxx-1,xxx-2,xxx-3,xxx-4,xxx-5,xxx-6,xxx-7,xxx-8,xxx-9,xxx-10,xxx-11,xxx-12,xxx-13,xxx-14,xxx-15,xxx-16,17,18,19,20,!21,!22,!23,!24,!25,!26,!27,!28']}
Failed to patch on /api/v1/playbooks/2175: {'labels': ['check:True', 'tags:all', 'subset:xxx-1,xxx-2,xxx-3,xxx-4,xxx-5,xxx-6,xxx-7,xxx-8,xxx-9,xxx-10,xxx-11,xxx-12,xxx-13,xxx-14,xxx-15,xxx-16,17,18,19,20,!21,!22,!23,!24,!25,!26,!27,!28']}
[WARNING]: Failure using method (v2_playbook_on_play_start) in callback plugin (<ansible.plugins.callback.ara_default.CallbackModule object at 0x7f1ace7322e8>): Expecting value: line 1 column 1 (char 0)
[WARNING]: Failure using method (v2_playbook_on_task_start) in callback plugin (<ansible.plugins.callback.ara_default.CallbackModule object at 0x7f1ace7322e8>): 'NoneType' object is not subscriptable
[WARNING]: Failure using method (v2_runner_on_skipped) in callback plugin (<ansible.plugins.callback.ara_default.CallbackModule object at 0x7f1ace7322e8>): 'NoneType' object is not subscriptable
```
I hope that helps.

Regards,

**Comment by rntbrgtt** [comment]
Hi David,

Thanks for the quick response and workaround to be used.

Regards,

**Comment by dmsimard** [comment]
I've sent a PR to truncate label names to 255 characters and reverted the migration that upped the limit to 512 characters to prevent impacting mysql support: https://github.com/ansible-community/ara/pull/437

The callback will print a warning when it truncates something so it is not done silently.

It isn't ideal but shouldn't result in exceptions if there are things longer than 255 characters now.

**Comment by dmsimard** [comment]
Closing this issue now that the fix has merged in the master branch. It'll be released in 1.6.0.
